kvmagent/kvmagent/plugins/vms/vm_host_file.py

```
@@ -0,0 +1,152 @@
+        # check: must use abs path
+        lr, out, le = bash.bash_roe("tar -tzf %s" % pipes.quote(tmp_tar_file))
+        if lr != 0:
+            raise RuntimeError("Failed to list tar file, stderr: %s" % (le))
+        for member in out.splitlines():
+            normalized = os.path.normpath(member)
+            if os.path.isabs(member) or normalized.startswith('..'):
+                raise ValueError("Unsafe tar entry detected: %s" % member)
+
+        items = os.listdir(extract_dir)
+        if not items:
+            raise ValueError("Tarball is empty, nothing to write to %s" % path)
```

Ruff (0.15.2)

[warning] Abstract `raise` to an inner function

(TRY301)